src/with-urql-client.tsx

```diff
@@ -76,12 +81,17 @@ function withUrqlClient<T = any, IP = any>(
     const displayName = Page.displayName || Page.name || 'Component';
     withUrql.displayName = `withUrqlClient(${displayName})`;
 
-    withUrql.getInitialProps = async (ctx: NextPageContext) => {
+    withUrql.getInitialProps = async (ctx: NextPageContext | AppContext) => {
       const { AppTree } = ctx;
 
+      const appCtx = (ctx as AppContext).ctx;
+      const isApp = !!appCtx;
+
       const opts =
-        typeof clientConfig === 'function' ? clientConfig(ctx) : clientConfig;
-      const [urqlClient, ssrCache] = initUrqlClient(opts);
+        typeof clientConfig === 'function'
+          ? clientConfig(isApp ? appCtx : (ctx as NextPageContext))
+          : clientConfig;
+      const [urqlClient, ssrCache] = initUrqlClient(opts, mergeExchanges);
```

Comment:
This was always missing `mergExchanges` 😬 If someone had a custom exchange running, it would never have gotten incorporated into the initial `Client` creation (the one that actually runs the prepass step). :yikes: